tests/lsp/test_signature_help.py

```diff
@@ -78,6 +80,7 @@ def test_capabilities(client_server):
 
 
 @ConfiguredLS.decorate()
+@pytest.mark.skip
```

Comment:
I haven't looked into it yet, but the changes in this PR seem to break the tests skipped in this file in such a way that they hang the entire test suite. The only error message I was able to get was something going wrong in the `exceptiongroup` package...
```
============================================================================================================= FAILURES ==============================================================================================================
______________________________________________________________________________________ test_signature_help_return_signature_help[ConfiguredLS] ______________________________________________________________________________________

exc = <class 'pygls.exceptions.JsonRpcInvalidParams'>, value = JsonRpcInvalidParams('Invalid Params'), tb = <traceback object at 0x7f708a6fee00>, limit = None, chain = True

    def format_exception(exc, /, value=_sentinel, tb=_sentinel, limit=None, \
                         chain=True):
        """Format a stack trace and the exception information.
    
        The arguments have the same meaning as the corresponding arguments
        to print_exception().  The return value is a list of strings, each
        ending in a newline and some containing internal newlines.  When
        these lines are concatenated and printed, exactly the same text is
        printed as does print_exception().
        """
        value, tb = _parse_value_tb(exc, value, tb)
        te = TracebackException(type(value), value, tb, limit=limit, compact=True)
>       return list(te.format(chain=chain))

/usr/lib64/python3.10/traceback.py:136: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
.env/lib64/python3.10/site-packages/exceptiongroup/_formatting.py:233: in traceback_exception_format
    yield from exc.exceptions[i].format(chain=chain, _ctx=_ctx)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <traceback.TracebackException object at 0x7f708a8ea020>

    def traceback_exception_format(self, *, chain=True, _ctx=None):
        if _ctx is None:
            _ctx = _ExceptionPrintContext()
    
        output = []
        exc = self
        if chain:
            while exc:
>               if exc.__cause__ is not None:
E               AttributeError: 'TracebackException' object has no attribute '__cause__'

.env/lib64/python3.10/site-packages/exceptiongroup/_formatting.py:169: AttributeError
```

Reply:
How do you know that these tests weren't originally skipped precisely because of the hanging? I mean, maybe your changes aren't actually affecting any behaviour here right?

Reply:
> How do you know that these tests weren't originally skipped precisely because of the hanging?

Because I added the skip :)

Reply:
😆